pkg/migration/migration.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+
+	"sigs.k8s.io/krew/pkg/constants"
+	"sigs.k8s.io/krew/pkg/environment"
+	"sigs.k8s.io/krew/pkg/index"
+	"sigs.k8s.io/krew/pkg/migration/oldenvironment"
+)
+
+const (
+	krewPluginName = "krew"
+)
+
+func IsMigrated(newPaths environment.Paths) bool {
+	_, err := os.Lstat(newPaths.InstallReceiptPath())
+	return err == nil
```

Comment:
I think some errors can be legit FS errors, although they're rare. using `os.IsNotExist(err)` is often better in these situations.

But since there's a better logic to detect this, don't worry.